source/components/GlobalNavigation/components/Notifications/NotificationsDataProvider.js

```diff
@@ -77,11 +77,10 @@ class NotificationsDataProvider extends React.Component {
         this.api
             .loadPage(pageLink)
             .then((response) => {
-                const mappedNotifications = this.mapNotifications(response.notifications);
+                const mappedNotifications = notifications.concat(this.mapNotifications(response.notifications));
```

Comment:
is this logic correct? it seems it'd duplicate notifications

Reply:
it should not.

`loadPage` is used for both first and following notification requests.
`firstPage` won't load twice because of `firstPageLoaded` field in state. 
`nextPage` request will return only new notifications

same logic lives in Ember https://github.com/Wikia/design-system/blob/IW-1022/app/models/wds-on-site-notifications/wds-on-site-notifications.js#L75-L85 so I assume there will be no dupes.